sdk/core/azure-core/src/samples/java/com/azure/core/http/rest/PagedIterableJavaDocCodeSnippets.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+import java.util.Iterator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Code snippets for {@link PagedIterable}
+ */
+public class PagedIterableJavaDocCodeSnippets {
+
+    /**Provides an example for iterate over each response using streamByPage function.**/
+    public void streamByPageSnippet() {
+
+        PagedFlux<Integer> pagedFlux = createAnInstance();
+        PagedIterable<Integer> pagedIterableResponse =  new PagedIterable<>(pagedFlux);
+
+        // BEGIN: com.azure.core.http.rest.pagedIterable.streamByPage
+        // process the streamByPage
+        pagedIterableResponse.streamByPage().forEach(resp -> {
+            if (resp.statusCode() == HttpURLConnection.HTTP_OK) {
+                System.out.printf("Response headers are %s. Url %s \n", resp.headers(), resp.request().url());
+                resp.items().forEach(value -> {
+                    System.out.printf("Response value is %d \n", value);
+                });
+            } else {
+                System.out.printf("Failed with status is %d and url is %s \n", resp.statusCode(), resp.request().url());
+            }
+        });
+        // END: com.azure.core.http.rest.pagedIterable.streamByPage
+    }
+
+    /**Provides an example for iterate over each response using iterableByPage function.**/
+    public void iterateByPageSnippet() {
+
+        PagedFlux<Integer> pagedFlux = createAnInstance();
+        PagedIterable<Integer> pagedIterableResponse =  new PagedIterable<>(pagedFlux);
+
+        // BEGIN: com.azure.core.http.rest.pagedIterable.iterableByPage
+        // process the iterableByPage
+        pagedIterableResponse.iterableByPage().forEach(resp -> {
+            if (resp.statusCode() == HttpURLConnection.HTTP_OK) {
+                System.out.printf("Response headers are %s. Url %s \n", resp.headers(), resp.request().url());
```

Comment:
IIRC, %n is preferred to \n when using string.format() because it produces a platform specific line ending.

Spotbugs entry:
```
FS: Format string should use %n rather than \n (VA_FORMAT_STRING_USES_NEWLINE)
This format string include a newline character (\n). In format strings, it is generally preferable better to use %n, which will produce the platform-specific line separator.
```